be1-go/popcha/server.go

```diff
@@ -0,0 +1,410 @@
+package popcha
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"github.com/aaronarduino/goqrsvg"
+	"github.com/ajstarks/svgo"
+	"github.com/boombuler/barcode/qr"
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"github.com/zitadel/oidc/v2/pkg/oidc"
+	"github.com/zitadel/oidc/v2/pkg/op"
+	"golang.org/x/xerrors"
+	"html/template"
+	"net/http"
+	"os"
+	"popstellar/hub"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	// formatting for auth request validation error
+	errValidAuthFormat = "Error while validating the auth request: %s"
+
+	//error message for invalid response type
+	errInvalidRepsonseType = "response type is invalid for the implicit flow"
```

Comment:
I already made this comment in my first review but there's this small typo here :)

Reply:
Oh my bad, i will correct that asap